test/fixtures/for-in/assign-enhanced-in/expected.js

```diff
@@ -0,0 +1,3 @@
+for (const k in obj = complexFunction()) {
+  const v = (obj = complexFunction())[k];
```

Comment:
Yes, this is part of the _ref change, and I expect the compiled code to look like the latter when done.